test/verify/machineslib.py

```diff
@@ -1436,6 +1436,61 @@ def installWithErrorTest(dialog):
+                    "cp /tmp/cert.pem /usr/local/share/ca-certificates/cert.crt",
+                    "update-ca-certificates"
+                ]
+            else:
+                cmds += [
+                    "cp /tmp/cert.pem /etc/pki/ca-trust/source/anchors/cert.pem",
+                    "update-ca-trust"
+                ]
+            self.machine.execute(" && ".join(cmds))
+
+            # Run https server with range option support. QEMU uses range option
+            # see: https://lists.gnu.org/archive/html/qemu-devel/2013-06/msg02661.html
+            # or
+            # https://github.com/qemu/qemu/blob/master/block/curl.c
+            #
+            # and on certain distribution supports only https (not http)
+            # see: block-drv-ro-whitelist option in qemu-kvm.spec for certain distribution
+            server = self.machine.spawn("cd /var/lib/libvirt && exec python3 /tmp/mock-range-server.py /tmp/server.pem", "httpsserver")
+            self.addCleanup(self.machine.execute, "kill {0}".format(server))
+
+        def fakeFedoraTree():
+            distro_tree_path = "/var/lib/libvirt/pub/archive/fedora/linux/releases/28/Server/x86_64/os/"
+            self.machine.execute("mkdir -p {0}".format(distro_tree_path))
+            self.machine.upload(["verify/files/fakefedoratree/.treeinfo", "verify/files/fakefedoratree/images"], distro_tree_path)
+            # borrow the kernel executable for the test server
+            self.machine.execute("cp /boot/vmlinuz-{0} {1}images/pxeboot/vmlinuz".format(self.machine.execute("uname -r").rstrip(), distro_tree_path))
+            self.machine.execute("chown -R 777 /var/lib/libvirt/pub")
+            self.restore_file("/etc/hosts")
+            self.machine.execute('echo "127.0.0.1 archive.fedoraproject.org" >> /etc/hosts')
+
+        def fakeOsDBInfo():
+            # Fake the osinfo-db data in order that it will allow spawn the installation - of course we don't expect it to succeed -
+            # we just need to check that the VM was spawned
+            fedora_28_xml = self.machine.execute("cat /usr/share/osinfo/os/fedoraproject.org/fedora-28.xml")
+            root = ET.fromstring(fedora_28_xml)
+            root.find('os').find('resources').find('minimum').find('ram').text = '134217750'
+            root.find('os').find('resources').find('minimum').find('storage').text = '134217750'
+            root.find('os').find('resources').find('recommended').find('ram').text = '268435500'
+            root.find('os').find('resources').find('recommended').find('storage').text = '268435500'
+            new_fedora_28_xml = ET.tostring(root)
+            self.machine.execute("echo \'{0}\' > /tmp/fedora-28.xml".format(str(new_fedora_28_xml, 'utf-8')))
+            self.machine.execute("mount -o bind /tmp/fedora-28.xml /usr/share/osinfo/os/fedoraproject.org/fedora-28.xml")
+            self.addCleanup(self.machine.execute, "umount /usr/share/osinfo/os/fedoraproject.org/fedora-28.xml || true")
```

Comment:
The `|| true` should not be necessary here. The test does not unmount the directory by itself.